satpy/composites/__init__.py

```diff
@@ -154,10 +153,21 @@ def _process_composite_config(self, composite_name, conf,
             prereqs = []
             for item in options.get(prereq_type, []):
                 if isinstance(item, dict):
-                    # we want this prerequisite to act as a query with
-                    # 'modifiers' being None otherwise it will be an empty
-                    # tuple
-                    item.setdefault('modifiers', None)
+                    # Handle in-line composites
+                    if 'compositor' in item:
+                        # Create an unique temporary name for the composite
+                        import uuid
+                        temp_comp_name = 'temp_' + str(uuid.uuid4())
+                        # Minimal composite config
+                        conf_tmp = {composite_type: {temp_comp_name: item}}
```

Comment:
Would it be possible to hash the composite arguments or something so it isn't unique? This way composites could be reused...eh maybe that's a bad idea. Actually that's the whole point of defining the dependencies not inline.